### PR TITLE
[react-native-win32] Add listitem accessibility role and accessibilityControls test example

### PR DESCRIPTION
**Pull request by PPatBoyd**
### Type of Change
New feature

### Why
List exists as an AccessibilityRole in react-native core, but win32 further clarifies the list contents with AccessibilityRole ListItem.  This role is not cross-platform, e.g. MacOS does not have a matching concept.

This addition expands support for additional accessibility scenarios.

### What
Added listitem to react-native-win32 type ARIARole.

## Screenshots
New accessibilityControls + List/ListItem test:
image

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/9902)

### CODE REVIEW

File: packages/@office-iss/react-native-win32-tester/src/js/examples-win32/Accessibility/AccessibilityExampleWin32.tsx [hunk 1]
```diff
@@ -223,9 +224,9 @@ const SelectionItemComponent: React.FunctionComponent<ISelectionItemComponentPro
         onAccessibilityAction={_onAccessibilityAction}
         onAccessibilityTap={_onPress}
       >
-        <Text>Item {props.value.toString()}</Text>
+        <TextWin32>{props.value.toString()}</TextWin32>
```
Comment (acoates-ms) [inline packages/@office-iss/react-native-win32-tester/src/js/examples-win32/Accessibility/AccessibilityExampleWin32.tsx:227]: If you are not doing anything win32 specific, I'd use the regular Text component.

**Comment by acoates-ms** [comment]
You'll need to run `yarn change` at the root.

**Comment by acoates-ms** [comment]
Have we looked at adding listitem to core? It is a standard Aria type.

And have you looked at the list we support in windows to see if there are any other obvious ones we should add to win32 too?
https://github.com/microsoft/react-native-windows/blob/67ae26c46f1be65d41151d2564f12fe04b736b41/vnext/Microsoft.ReactNative/Views/FrameworkElementViewManager.cpp#L316

**Comment by PPatBoyd** [comment]
> Have we looked at adding listitem to core? It is a standard Aria type.
> 
> And have you looked at the list we support in windows to see if there are any other obvious ones we should add to win32 too?
> 
> https://github.com/microsoft/react-native-windows/blob/67ae26c46f1be65d41151d2564f12fe04b736b41/vnext/Microsoft.ReactNative/Views/FrameworkElementViewManager.cpp#L316

For adding to core I've thought about it and looked into comparable props, but I haven't fully rationalized yet what's needed for the other platforms -- AppKit (iOS, macOS) only has a `List` and no ListItem concept in in NSAccessibilityRole set, and I haven't dug through android accessibility enough to really understand it -- seemingly the CollectionInfo and CollectionItemInfo classes align [AccessibilityNodeInfo classes](https://developer.android.com/reference/android/view/accessibility/AccessibilityNodeInfo); there's a recent PR that was introduced and backed out of core for Android [To Support Collection Item Info in lists and grids](https://github.com/facebook/react-native/commit/dd6325bafe1a539d348f3710e717a6344576b859) -- that's a new find for me today refreshing myself on the other platforms. With some ongoing work I should have a better understanding for AppKit (iOS/MacOS); seemingly Windows shouldn't take much but I haven't spent time in the pure RNW environment.

**Comment by acoates-ms** [comment]
Can you sign the CLA?  -- Not sure what's up with it - it seems to have forgotten everyone... or it changed or something.

**Review by acoates-ms — APPROVED** [review]

**Pull request merged**